tslearn/nonmyopic.py

```diff
@@ -0,0 +1,370 @@
+        associated classes. A classifier is fit for each time stamp above a minimum time stamp that is an attribute of
+        the class of the model. Then some probabilities are computed using clusters already computed.
+        This function should be divided among 2 or 3 functions.
+
+        Parameters
+        ----------
+        X: Array-like
+            a dataset of time series
+        Y: vector
+            the associated classes of the series from X_train
+        """
+        classes_y = np.unique(Y)
+        self.cluster_ = TimeSeriesKMeans(n_clusters=self.n_clusters)
+        self.classifier_ = []
+        self.clusters_ = {}
+        self.pyhatyck_ = {}
+        self.indice_ck_ = []
+        self.__n_classes_ = len(classes_y)
+        self.__len_X_ = X.shape[1]
+        c_k = self.cluster_.fit_predict((to_time_series_dataset(X)))
+        mid_X = int(X.shape[0] / 2)
+        X1 = X[:mid_X]
+        Y1 = Y[:mid_X]
+        X2 = X[mid_X:]
+        Y2 = Y[mid_X:]
+        c_k2 = c_k[mid_X:]
+        vector_of_ones = np.ones((len(X[:]),))
+        self.pyck_ = coo_matrix((vector_of_ones, (Y, c_k)), shape=(self.__n_classes_, self.n_clusters)).toarray()
+        for k in range(0, self.n_clusters):
+            self.clusters_["ck_cm{0}".format(k)] = []
+            self.pyhatyck_["pyhatycks{0}".format(k)] = []
+            self.indice_ck_.append(np.where(c_k == k))
+            current_sum = self.pyck_[:, k].sum()
+            for current_classe in range(0, self.__n_classes_):
+                self.pyck_[current_classe, k] = self.pyck_[current_classe, k] / current_sum
+
+        for t in range(self.minimum_time_stamp, X.shape[1] + 1):
+            self.classifier_.append(
+                self.classifier(
+                    solver=self.solver,
+                    hidden_layer_sizes=self.hidden_layer_sizes,
+                    random_state=self.random_state,
+                    max_iter=self.maximum_iteration,
+                )
+            )
+            self.classifier_[-1].fit(X1[:, :t], Y1)
+            for k in range(0, self.n_clusters):
+                index_cluster = np.where(c_k2 == k)
+                if len(index_cluster[0]) != 0:
+                    X2_current_cluster = np.squeeze(X2[index_cluster, :t], axis=0)
+                    Y2_current_cluster = Y2[tuple(index_cluster)]
+                    Y2_hat = self.classifier_[-1].predict(X2_current_cluster[:, :t])
+                    self.clusters_["ck_cm{0}".format(k)].append(
+                        confusion_matrix(Y2_current_cluster, Y2_hat, labels=classes_y)
+                    )
+                    column_sum = self.clusters_["ck_cm{0}".format(k)][
+                        -1
+                    ].sum(1)
+                    for current_class in range(0, self.__n_classes_):
+                        if column_sum[current_class] == 0:
+                            column_sum[current_class] = 1
+                    matrix_proba_current_cluster_class = np.asarray(
```

Comment:
This is not readable as is. Probably a limitation of `black`, please reformat it to be more human-eye friendly.